build/source/engine/computFlux.f90

```diff
@@ -0,0 +1,854 @@
+                 dHydCond_dMatric,                       & ! intent(out):   derivative in hydraulic conductivity w.r.t matric head (s-1)
+                 ! output: fluxes
+                 scalarInfiltration,                     & ! intent(out):   surface infiltration rate (m s-1) -- controls on infiltration only computed for iter==1
+                 iLayerLiqFluxSoil,                      & ! intent(out):   liquid fluxes at layer interfaces (m s-1)
+                 mLayerTranspire,                        & ! intent(out):   transpiration loss from each soil layer (m s-1)
+                 mLayerHydCond,                          & ! intent(out):   hydraulic conductivity in each layer (m s-1)
+                 ! output: derivatives in fluxes w.r.t. state variables -- matric head or volumetric lquid water -- in the layer above and layer below (m s-1 or s-1)
+                 dq_dHydStateAbove,                      & ! intent(out):   derivatives in the flux w.r.t. matric head in the layer above (s-1)
+                 dq_dHydStateBelow,                      & ! intent(out):   derivatives in the flux w.r.t. matric head in the layer below (s-1)
+                 ! output: derivatives in fluxes w.r.t. energy state variables -- now just temperature -- in the layer above and layer below (m s-1 K-1)
+                 dq_dNrgStateAbove,                      & ! intent(out):   derivatives in the flux w.r.t. temperature in the layer above (m s-1 K-1)
+                 dq_dNrgStateBelow,                      & ! intent(out):   derivatives in the flux w.r.t. temperature in the layer below (m s-1 K-1)
+                 ! output: error control
+                 err,cmessage)                             ! intent(out): error control
+ if(err/=0)then; message=trim(message)//trim(cmessage); return; endif
+ ! calculate net liquid water fluxes for each soil layer (s-1)
+ do iLayer=1,nSoil
+  mLayerLiqFluxSoil(iLayer) = -(iLayerLiqFluxSoil(iLayer) - iLayerLiqFluxSoil(iLayer-1))/mLayerDepth(iLayer+nSnow)
+ end do
+ ! calculate the soil control on infiltration
+ if(nSnow==0) then
+  ! * case of infiltration into soil
+  if(scalarMaxInfilRate > scalarRainPlusMelt)then  ! infiltration is not rate-limited
+   scalarSoilControl = (1._dp - scalarFrozenArea)*scalarInfilArea
+  else
+   scalarSoilControl = 0._dp  ! (scalarRainPlusMelt exceeds maximum infiltration rate
+  endif
+ else
+  ! * case of infiltration into snow
+  scalarSoilControl = 1._dp
+ endif
+
+ ! expand derivatives to the total water matric potential
+ dq_dHydStateAbove(1:nSoil)   = dq_dHydStateAbove(1:nSoil)  *dPsiLiq_dPsi0(1:nSoil)
+ dq_dHydStateBelow(0:nSoil-1) = dq_dHydStateBelow(0:nSoil-1)*dPsiLiq_dPsi0(1:nSoil)
+
+ ! *****
+ ! * CALCULATE THE GROUNDWATER FLOW...
+ ! ************************************
+
+ ! set baseflow fluxes to zero if the baseflow routine is not used
+ if(local_ixGroundwater/=qbaseTopmodel)then
```

Comment:
True. Though it is inefficient to initialize to zero when the method is active. Original code retained.